src/v/cluster/partition_manager.cc

```diff
@@ -77,6 +77,28 @@ ss::future<> partition_manager::remove(const model::ntp& ntp) {
       .finally([partition] {}); // in the end remove partition
 }
 
+ss::future<> partition_manager::shutdown(const model::ntp& ntp) {
+    auto partition = get(ntp);
+
+    if (!partition) {
+        return ss::make_exception_future<>(std::invalid_argument(fmt::format(
+          "Can not shutdown partition. NTP {} is not present in partition "
+          "manager",
+          ntp)));
+    }
+    auto group_id = partition->group();
+
+    // remove partition from ntp & raft tables
+    _ntp_table.erase(ntp);
+    _raft_table.erase(group_id);
+
+    return _raft_manager.local()
```

Comment:
This should catch exceptions and crash if error with a vassert() so we don’t loose context. 

If something were to happen in *any* sub component, we cannot continue.

Reply:
@mmaslankaprv were you planning to do this?